Sources/PackageConfig/Package.swift

```diff
@@ -100,12 +100,35 @@ enum Package {
 		let fileManager = FileManager.default
 
 		let swiftPMDir = swiftPath.replacingOccurrences(of: "bin/swiftc", with: "lib/swift/pm")
-		let versions = try! fileManager.contentsOfDirectory(atPath: swiftPMDir)
-		#warning("TODO: handle //swift-tools-version:4.2 declarations")
-		let latestSPM = versions.sorted().last!
-		let libraryPathSPM = swiftPMDir + "/" + latestSPM
+		let versions = try! fileManager.contentsOfDirectory(atPath: swiftPMDir).filter { $0 != "llbuild" }
+
+        let swiftToolsVersion = getSwiftToolsVersion()
+        let spmDir = versions.contains(swiftToolsVersion) ? swiftToolsVersion : versions.sorted().last!
+		let libraryPathSPM = swiftPMDir + "/" + spmDir
 
 		debugLog("Using SPM version: \(libraryPathSPM)")
 		return ["-L", libraryPathSPM, "-I", libraryPathSPM, "-lPackageDescription"]
 	}
+    
+    static private func getSwiftToolsVersion() -> String {
```

Comment:
I think you could use `String?`, IMO `nil` is more clear than `""` for when you don't have a version